cdap-docs/reference-manual/source/http-restful-api/metadata.rst

```diff
@@ -0,0 +1,529 @@
+
+  GET /v3/namespaces/<namespace>/<element-type>/<element-id>/metadata/tags
+  
+or, for a particular program of a specific application::
+
+  GET /v3/namespaces/<namespace>/apps/<app-id>/<program-type>/<program-id>/metadata/tags
+
+with the metadata tags returned as a JSON string in the return body::
+
+  ["tag1", "tag2"]
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``<namespace>``
+     - Namespace ID
+   * - ``<element-type>``
+     - One of ``apps``, ``datasets``, or ``streams``
+   * - ``<element-id>``
+     - Name of the element
+   * - ``<app-id>``
+     - Name of the application
+   * - ``<program-type>``
+     - One of ``flows``, ``mapreduce``, ``spark``, ``workflows``, ``services``, or ``workers``
+   * - ``<program-id>``
+     - Name of the program
+
+.. rubric:: HTTP Responses
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Status Codes
+     - Description
+   * - ``200 OK``
+     - The properties requested were returned as a JSON string in the body of the response
+   * - ``404 NOT FOUND``
+     - The entity (element or program) for which properties are being retreived was not found
+     
+     
+Removing Tags
+-------------
+To delete all user metadata tags for an application, dataset, or stream, submit an
+HTTP DELETE request::
+
+  DELETE /v3/namespaces/<namespace>/<element-type>/<element-id>/metadata/tags
+  
+or, for all user metadata tags of a particular program of a specific application::
+
+  DELETE /v3/namespaces/<namespace>/apps/<app-id>/<program-type>/<program-id>/metadata/tags
+
+To delete a specific user metadata tag for an application, dataset, or stream, submit
+an HTTP DELETE request with the tag::
+
+  DELETE /v3/namespaces/<namespace>/<element-type>/<element-id>/metadata/properties/<tag>
+  
+or, for a particular user metadata tag of a program of a specific application::
+
+  DELETE /v3/namespaces/<namespace>/apps/<app-id>/<program-type>/<program-id>/metadata/properties/<tag>
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``<namespace>``
+     - Namespace ID
+   * - ``<element-type>``
+     - One of ``apps``, ``datasets``, or ``streams``
+   * - ``<element-id>``
+     - Name of the element
+   * - ``<app-id>``
+     - Name of the application
+   * - ``<program-type>``
+     - One of ``flows``, ``mapreduce``, ``spark``, ``workflows``, ``services``, or ``workers``
+   * - ``<program-id>``
+     - Name of the program
+   * - ``<tag>``
+     - Metadata tag
+
+.. rubric:: HTTP Responses
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Status Codes
+     - Description
+   * - ``200 OK``
+     - The method was successfully called, and the tags were deleted, or in the case of a
+       specific tag, was either deleted or the tag was not present
+   * - ``404 NOT FOUND``
+     - The entity (element or program) for which tags are being deleted was not found
```

Comment:
We use element and entity interchangeably. It can be confusing in some contexts. We should stick to one I think. What do you say?